http-client/src/main/java/io/micronaut/http/client/netty/DefaultHttpClient.java

```diff
@@ -1491,181 +1454,360 @@ private <I> Publisher<URI> resolveURI(io.micronaut.http.HttpRequest<I> request,
 
-        Promise<MutableHttpResponse<?>> responsePromise = poolHandle.channel.eventLoop().newPromise();
-        ChannelPipeline pipeline = poolHandle.channel.pipeline();
-        pipeline.addLast(ChannelPipelineCustomizer.HANDLER_MICRONAUT_HTTP_RESPONSE, new Http1ResponseHandler(new StreamHttpResponseListener(responsePromise, parentRequest, request, poolHandle, sse)));
-        poolHandle.notifyRequestPipelineBuilt();
+        FilterRunner.sortReverse(filters);
 
-        if (log.isDebugEnabled()) {
-            debugRequest(request.getUri(), nettyRequest);
+        FilterRunner runner = new FilterRunner(filters) {
+            @Override
+            protected ExecutionFlow<HttpResponse<?>> provideResponse(io.micronaut.http.HttpRequest<?> request, PropagatedContext propagatedContext) {
+                try {
+                    try (PropagatedContext.Scope ignore = propagatedContext.propagate()) {
+                        return ReactiveExecutionFlow.fromPublisher(Mono.from(sendRequestWithRedirectsNoFilter(
+                            parentRequest,
+                            blockHint,
+                            MutableHttpRequestWrapper.wrapIfNecessary(conversionService, request),
+                            readResponse
+                        )));
+                    }
+                } catch (Throwable e) {
+                    return ExecutionFlow.error(e);
+                }
+            }
+        };
+        Mono<HttpResponse<?>> responseMono = Mono.from(ReactiveExecutionFlow.fromFlow(runner.run(request)).toPublisher());
+        if (parentRequest != null) {
+            responseMono = responseMono.contextWrite(c -> {
+                // existing entry takes precedence. The parentRequest is derived from a thread
+                // local, and is more likely to be wrong than any reactive context we are fed.
+                if (c.hasKey(ServerRequestContext.KEY)) {
+                    return c;
+                } else {
+                    return c.put(ServerRequestContext.KEY, parentRequest);
+                }
+            });
         }
+        return responseMono;
+    }
 
-        if (log.isTraceEnabled()) {
-            traceRequest(request, nettyRequest);
+    private Mono<HttpResponse<?>> sendRequestWithRedirectsNoFilter(
+        io.micronaut.http.HttpRequest<?> parentRequest,
+        @Nullable BlockHint blockHint,
+        MutableHttpRequest<?> request,
+        BiFunction<MutableHttpRequest<?>, NettyClientByteBodyResponse, ? extends Mono<? extends HttpResponse<?>>> readResponse
+    ) {
+        RequestKey requestKey;
+        try {
+            requestKey = new RequestKey(this, request.getUri());
+        } catch (Exception e) {
+            return Mono.error(e);
         }
+        // first: connect
+        return connectionManager.connect(requestKey, blockHint)
+            .flatMap(poolHandle -> {
```

Comment:
Please extract a new methoda

Reply:
i dont agree, this code is basically imperative